### ISSUE TITLE
Single FastCast for both receiving and offering

### ISSUE THREAD

**Issue by dmart28** [RuedigerMoeller/fast-cast#3]
Hi there,

Why I can only use FastCast instance topic either for sending or receiving, but not both?

```
public void subscribe( SubscriberConf subsConf, FCSubscriber subscriber ) {
        ....
        if ( topicEntry.getPublisherConf() != null ) {
            throw new RuntimeException("already a sender registered at "+subsConf.getTopicId());
        }
```

Why this condition here, how publisher intersects with subscriber after all? Appreciate for answer.

**Comment by dmart28** [comment]
OK, It seems registering subscriber first and publisher next works as workaround. But still, some explanation about such internal logic is very helpful.

**Comment by RuedigerMoeller** [comment]
Hi there, that's probably a bug, it was somehow never tested the other way around. 
In fact a sender always listens as it has to listen for retransmission requests. So in case there is no receiver, it installs one which makes the sanity check fail when later on adding a receiver :)

Thanks for the report.

**Comment by RuedigerMoeller** [comment]
it should be fixed ofc

**Comment by RuedigerMoeller** [comment]
actually lost messages as reported in your second issue might result from a sender not receiving retransmission packets .. will check tomorrow evening.

**Comment by RuedigerMoeller** [comment]
fixed with 3.10. cut+paste fail
